### ISSUE TITLE
[feature] Support cross-validation and train-val splits from training set

### ISSUE THREAD

**Issue by apsdehal** [facebookresearch/mmf#450]
## 🚀 Feature

MMF should have a feature to split training set in training and validation set randomly by some percentage and thus allowing cross-validation possible.

## Motivation

A lot of datasets only come with a training and testing split which doesn't fit in the paradigm of MMF which requires a validation set to be present. This limitation doesn't also allow the users to use validation set for training as well and use cross-validation for selecting the best model. 

## Pitch

A option in dataset config which can enable validation split from training split. This option will basically take a percentage which is how much the training set is to be split into validation. If it is -1, MMF should expect the validation set to be present there by default. sklearn has good utilities on split a set into splits randomly.

Note: If you are a bootcamper and working on this task, please assign the internal task to yourself: T71146572

**Comment by ytsheng** [comment]
Fixed with #470
